script/testing/oltpbench/README.md

```diff
@@ -1,18 +1,94 @@
+    --config-file=configs/end_to_end_debug/tpcc.json \
+    --build-type=debug
+  ```
+- End-to-end performance
+  ```bash
+  cd noisepage/script/testing/oltpbench
+  # To run the TPCC oltpbench test for end-to-end performance with wal to be stored in ramdisk on release version
+  python3 run_oltpbench.py \
+    --config-file=configs/end_to_end_performance/tpcc_wal_ramdisk.json \
+    --build-type=release
+  ```
+- Nightly performance
+  ```bash
+  cd noisepage/script/testing/oltpbench
+  # To run all the oltpbench test case for nightly performance with wal disabled on release version
+  python3 run_oltpbench.py \
+    --config-file=configs/nightly/nightly_wal_disabled.json \
+    --build-type=release
+  ```
+
+### Advanced options
+For more configurations, there are 2 ways
+- Via config files
+  - You can refer to the [config files](https://github.com/cmu-db/noisepage/blob/master/script/testing/oltpbench/README.md#config-files) section below for more details
+- Via command line options
+  - You can use `-h` or `--help` command line option for more details
+  - *In theory, if the same config are conflicting between config files and command line options, the config from __command line option should prevail__*
+
+## Test workflow
+- Start by invoking `run_oltpbench.py`
+  - Load the config file passed in and start an instance of the database with any server_args specified in the config
+  - Create a test suite for the whole config file and runs the test suite
+  - Create a `TestOLTPBench` object for the test execution
+    - Pass command line options and configs in to the constructor
+    - Try to locate the NoisePage binary
+  - Run the test suite by calling `.run()` function of `TestOLTPBench`
+- Run pre suite tasks: `.run_pre_suite()` function of `TestOLTPBench`
+  - Clean the possible residual local [oltpbench](https://github.com/oltpbenchmark/oltpbench) workspace
+  - Download and install the [oltpbench](https://github.com/oltpbenchmark/oltpbench) from GitHub
+  - [optional] *Checkout to the specified branch*
+- Iterate through all the test cases
+  - (Re)start the NoisePage DB process as a Python subprocess
+    - Kill all the lingering processes on the NoisePage port
+    - Start the NoisePage DB process
+  - Run pre test case tasks: `.run_pre_test()` function of `TestCaseOLTPBench`
+    - Create the database and tables for the OLTP benchmark specified
+    - Load the data to tables
+  - Run the test case command as a subprocess
+    - Collect the memory info by using the [PeriodicTask](https://github.com/cmu-db/noisepage/blob/master/script/testing/util/periodic_task.py) in a separate thread
+      - Collect `RSS` and `VMS` by default
+      - Collect every `5` seconds by default
+      - The memory info is stored in a Python dictionary in memory in runtime
+  - Run post test case tasks: `.run_post_test()` function of `TestCaseOLTPBench`
+    - If it is part of the Jenkins nightly build, the result results should be stored
```

Comment:
Instead of saying "if part of Nightly" let's change it to if `publish-results`, 'publish-username`, and `publish-password` arguments are supplied...

Reply:
done